pkg/commands/patch/patch_parser.go

```diff
@@ -183,16 +184,21 @@ func parsePatch(patch string) ([]int, []int, []*PatchLine) {
 
 // Render returns the coloured string of the diff with any selected lines highlighted
 func (p *PatchParser) Render(firstLineIndex int, lastLineIndex int, incLineIndices []int) string {
-	renderedLines := make([]string, len(p.PatchLines))
-	for index, patchLine := range p.PatchLines {
-		selected := index >= firstLineIndex && index <= lastLineIndex
-		included := utils.IncludesInt(incLineIndices, index)
-		renderedLines[index] = patchLine.render(selected, included)
-	}
-	result := strings.Join(renderedLines, "\n")
-	if strings.TrimSpace(utils.Decolorise(result)) == "" {
```

Comment:
unrelated to the refactor: this Decolorise call was a little expensive so I've removed it